src/twisted/python/_setup.py

```diff
@@ -96,6 +96,7 @@
         'pyasn1',
         'cryptography >= 0.9.1',
         'appdirs >= 1.4.0',
+        'pynacl >= 1.0.1',
```

Comment:
Additional cryptographic dependencies are often met with high levels of scrutiny.  If possible, I'd really like to see this implemented in a way which makes pynacl an _optional_ dependency.

This doesn't necessarily mean that the entry in `setup.py` needs to go away, but it strikes me that "cryptography yes, pynacl no" might end up being a configuration someone cares about.